lib/server.js

```diff
@@ -67,6 +67,10 @@ function CoAPServer(options, listener) {
     this._options.piggybackReplyMs = parameters.piggybackReplyMs
   }
 
+  if (!this._options.block2CacheExpiryMs || !isNumeric(this._options.block2CacheExpiryMs)) {
```

Comment:
I just have some general comments & questions; I don't know enough about this bit of the protocol to speak to specifics there.

If you're going to bother checking it's a number, you may want to check that it's also a *positive* number, and probably an integer (since we're talking about milliseconds).  Also, a couple questions:

- What happens if the number is `0`?  
- What is the relationship between this value and `exchangeLifetime`?  Would there be any sort of weirdness if it was set above or below some threshold?
- Why is the default value (line 71 below) what it is?